perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2714,6 +2716,23 @@ private void categorizeMembersForLightweightSynchronization(PerunSession sess, G
 					candidate.setAttributes(new HashMap<>());
 				}
 			} catch (UserExtSourceNotExistsException | UserNotExistsException ex) {
+				List<UserExtSource> additionalUserExtSources = Utils.extractAdditionalUserExtSources(sess, subjectFromLoginSource);
```

Comment:
There are some duplicities in these lines of code. Try to rewrite it without the duplicities.